docs/kbs_protocol.md

```diff
@@ -0,0 +1,276 @@
+However, Universal-KBS do not intend to apply these projects directly, nor will we design a new 
+similar implementation, because it is inevitable to make some deep-seated adaptation to the 
+existing transmission module at the server and client. We hope that the intrusion of 
+Universal-KBS into existing transmission modes will be as few as possible, so that Universal-KBS 
+can be applied in more and broader scenarios.
+
+Therefore, in Universal-KBS, HW-TEE attestation process is the semantics of the application 
+layer, which is defined as a simple, universal and extensible 
+"Request-Challenge-Attestation-Response" (RACR) protocol. The temporary asymmetric key generated 
+by HW-TEE is used to encrypt the response payload, and the token mechanism is used to avoid the 
+performance problems caused by multiple attestation.
+
+In order to ensure the ease of use and security completeness of Universal-KBS, we will use HTTPS 
+as the default transmission method to carry the application layer semantics designed in this 
+document. This is because HTTPS provides KBC with a means to authenticate KBS identity, which 
+effectively avoids malicious attackers from hijacking KBS address to impersonate KBS and deceive KBC.
+
+# RCAR semantics
+
+The semantics of attestation defined by Universal-KBS is a simple and extensible four-step model, 
+which uses JSON structure to organize information. As follows:
+
+1. **Request**: The KBC actively sends to KBS, calls the service API provided by KBS to request resources or services.
+2. **Challenge**: After receiving the request, KBS returns a challenge to KBC and asks KBC to 
+send evidence to prove that its environment (HW-TEE) is safe and reliable.
+3. **Attestation**: KBC sends the evidence and the newly generated public key in HW-TEE to KBS.
+4. **Response**: After the KBS verified the evidence, it returns the API output of the requested 
+service and a token to KBC, where the API output payload is protected by the public key in step 3. 
+Within the valid time of the token, KBC can directly request resources or services of KBS by 
+virtue of the token without step 2 and step 3.
+
+## Request
+
+The payload format of the request is as follows:
+
+(Note that the `/*...*/` comments are not valid in JSON, and must not be used in real message.)
+
+```json
+{
+    "request": {
+	    /* Protocol version number used by KBC */
+	    "protocol-version": "0.1.0",
+	    /* Types of HW-TEE platforms where KBC is located, such as "TDX", "SEV-SNP", etc. */
+	    "tee": "",
+	    /* Access token to avoid multiple attempts triggered by consecutive requests. It can be set to blank. */
+	    "token": "",
+	    /* The service API requested to be called. The supported API types are defined and published by KBS. */
+	    "api": "",
+	    /* API input parameters. */
+	    "api-input": {},
```

Comment:
This is indeed a risk point worth considering, but it should depend on the security level we adopt when implementing KBS, because confidential computing itself does not provide availability protection. We can remember this risk point, but it may not be the focus of this document.

Reply:
Yeah. We probably will just want some time limits on how long the KBS will store a request if it never gets an attestation. That is beyond the scope of the document but let's try to remember.